clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -2252,7 +2252,7 @@ void Parser::ParseBaseClause(Decl *ClassDecl) {
   while (true) {
     // Parse a base-specifier.
     BaseResult Result = ParseBaseSpecifier(ClassDecl);
-    if (Result.isInvalid()) {
+    if (!Result.isUsable()) {
```

Comment:
I am a bit surprised we still don't want the `BaseInfo.push_back(Result.get())` in the else branch in the good case. Why does not having this not change behavior?